clumper/clump.py

```diff
@@ -62,6 +63,56 @@ def read_json(cls, path):
             return Clumper(data)
         return Clumper(json.loads(pathlib.Path(path).read_text()))
 
+    @classmethod
+    def read_csv(cls, path, delimiter=",", fieldnames=None, nrows=None):
+        """
+        Reads in a csv file. Can also read files from url.
+
+        Parameters
+        ----------
+        path : filename or url
+        delimiter: must be a single character. `,` is the default.
+        fieldnames: If `None`, the first row of the csv is assumed as the header.
+        nrows: Number of rows to read in. Useful when reading large files. If `None`, all rows are read.
+
+        Usage:
+
+        ```python
+        from clumper import Clumper
+
+        clump = Clumper.read_csv("tests/iris.csv")
+        assert len(clump) == 150
+
+        clump = Clumper.read_csv("tests/iris.csv", nrows = 30)
+        assert len(clump) == 30
+
+        clump = Clumper.read_csv("https://calmcode.io/datasets/monopoly.csv")
+        assert len(clump) == 22
+        ```
+        """
+        if path.startswith("https:") or path.startswith("http:"):
+            with urllib.request.urlopen(path) as resp:
+
+                if fieldnames is not None:
+                    fieldnames = fieldnames
+                else:
+                    fieldnames = resp.readline().decode().strip().split(",")
+
+                # this section allows us to chunk the rows, if nrows is supplied
+                if nrows is not None:
+                    _, body = zip(
+                        *it.takewhile(lambda line: line[0] < nrows, enumerate(resp))
```

Comment:
Do you really need itertools here? You might also be able to do this with a list comprehension.

Reply:
Yes, a list comprehension can do this, but, correct me if I am wrong, the list comprehension will run through the list, whereas the takewhile breaks off once the condition fails.

Reply:
Ignore the earlier comment. `islice` does the job better and faster.